### ISSUE TITLE
REST API don't return valid JSON if fatal error triggerd

### ISSUE THREAD

**Issue by mesilov** [amplifr/amplifr-php#9]
Expect: 
- valid JSON with error description
- 5xx HTTP-code 

now we see:
`
[2016-09-30 00:51:36] amplifr.ERROR: network error, http code: 500

raw_response":"<!DOCTYPE html>\n<html>\n  <head>\n    <script src=\"https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js\"></script>\n    <script src=\"https://ajax.googleapis.com/ajax/libs/jqueryui/1.11.4/jquery-ui.min.js\"></script>\n    <link href=\"https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,400,600\" media=\"screen\" rel=\"stylesheet\" />\n    <link href=\"https://maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css\" media=\"screen\" rel=\"stylesheet\" />\n\n    <title>Amplifr / 500</title>\n\n    <style>\n      *{-moz-box-sizing:border-box;-webkit-box-sizing:border-box;box-sizing:border-box}html,body,div,span,object,iframe,h1,h2,h3,h4,h5,h6,p,blockquote,pre,abbr,address,cite,code,del,dfn,em,img,ins,kbd,q,samp,small,strong,sub,sup,var,b,i,dl,dt,dd,ol,ul,li,fieldset,form,label,legend,caption,article,aside,canvas,details,figcaption,figure,footer,header,hgroup,menu,nav,section,summary,time,mark,audio,video{margin:0;padding:0;border:0;outline:0;vertical-align:baseline;background:transparent}article,aside,details,figcaption,figure,footer,header,hgroup,nav,section{display:block}html{font-size:16px;line-height:24px;width:100%;height:100%;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%;overflow-y:scroll;overflow-x:hidden}img{vertical-align:middle;max-width:100%;height:auto;border:0;-ms-interpolation-mode:bicubic}body{min-height:100%;-webkit-font-smoothing:subpixel-antialiased}.clearfix{clear:both;zoom:1}.clearfix:before,.clearfix:after{content:&quot;\0020&quot;;display:block;height:0;visibility:hidden}.clearfix:after{clear:both}\n\n    </style>\n    <style>\n  .plain.error-page-wrapper {\n    font-family: 'Source Sans Pro', sans-serif;\n    background-color:#6355bc;\n    padding:0 5%;\n    position:relative;\n  }\n\n  .plain.error-page-wrapper .content-container {\n    -webkit-transition: left .5s ease-out, opacity .5s ease-out;\n    -moz-transition: left .5s ease-out, opacity .5s ease-out;\n    -ms-transition: left .5s ease-out, opacity .5s ease-out;\n    -o-transition: left .5s ease-out, opacity .5s ease-out;\n    transition: left .5s ease-out, opacity .5s ease-out;\n    max-width:400px;\n    position:relative;\n    left:-30px;\n    opacity:0;\n  }\n\n  .plain.error-page-wrapper .content-container.in {\n    left: 0px;\n    opacity:1;\n  }\n\n  .plain.error-page-wrapper .head-line {\n    transition: color .2s linear;\n    font-size:48px;\n    line-height:60px;\n    color:rgba(255,255,255,.2);\n    letter-spacing: -1px;\n    margin-bottom: 5px;\n  }\n  .plain.error-page-wrapper .subheader {\n    transition: color .2s linear;\n    font-size:36px;\n    line-height:46px;\n    color:#fff;\n  }\n  .plain.error-page-wrapper hr {\n    height:1px;\n    background-color: rgba(255,255,255,.2);\n    border:none;\n    width:250px;\n    margin:35px 0;\n  }\n  .plain.error-page-wrapper .context {\n    transition: color .2s linear;\n    font-size:18px;\n    line-height:27px;\n    color:#fff;\n  }\n  .plain.error-page-wrapper .context p {\n    margin:0;\n  }\n  .plain.error-page-wrapper .context p:nth-child(n+2) {\n    margin-top:12px;\n  }\n  .plain.error-page-wrapper .buttons-container {\n    margin-top: 45px;\n    overflow: hidden;\n  }\n  .plain.error-page-wrapper .buttons-container a {\n    transition: color .2s linear, border-color .2s linear;\n    font-size:14px;\n    text-transform: uppercase;\n    text-decoration: none;\n    color:#fff;\n    border:2px solid white;\n    border-radius: 99px;\n    padding:8px 30px 9px;\n    display: inline-block;\n    float:left;\n  }\n  .plain.error-page-wrapper .buttons-container a:hover {\n    background-color:rgba(255,255,255,.05);\n  }\n  .plain.error-page-wrapper .buttons-container a:first-child {\n    margin-right:25px;\n  }\n\n  @media screen and (max-width: 485px) {\n    .plain.error-page-wrapper .header {\n      font-size:36px;\n     }\n    .plain.error-page-wrapper .subheader {\n      font-size:27px;\n      line-height:38px;\n     }\n    .plain.error-page-wrapper hr {\n      width:185px;\n      margin:25px 0;\n     }\n\n    .plain.error-page-wrapper .context {\n      font-size:16px;\n      line-height: 24px;\n     }\n    .plain.error-page-wrapper .buttons-container {\n      margin-top:35px;\n    }\n\n    .plain.error-page-wrapper .buttons-container a {\n      font-size:13px;\n      padding:8px 0 7px;\n      width:45%;\n      text-align: center;\n    }\n    .plain.error-page-wrapper .buttons-container a:first-child {\n      margin-right:10%;\n    }\n  }\n</style>\n    <style>\n\n    .background-color {\n      background-color: rgba(235, 41, 92, 1) !important;\n    }\n\n\n    .primary-text-color {\n      color: #FFFFFF !important;\n    }\n\n    .secondary-text-color {\n      color: #FFFFFF !important;\n    }\n\n\n\n\n    .border-button {\n      color: #FFFFFF !important;\n      border-color: #FFFFFF !important;\n    }\n    .button {\n      background-color: #FFFFFF !important;\n      color:  !important;\n    }\n\n\n</style>\n  </head>\n  <body class=\"plain error-page-wrapper background-color background-image\">\n\n    \n    <noscript><iframe src=\"//www.googletagmanager.com/ns.html?id=GTM-MM887X\"\n        height=\"0\" width=\"0\" style=\"display:none;visibility:hidden\"></iframe></noscript>\n    <script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':\n      new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],\n      j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=\n      '//www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);\n      })(window,document,'script','dataLayer','GTM-MM887X');</script>\n    \n\n    <div class=\"content-container\">\n\t<div class=\"head-line secondary-text-color\">\n\t\t500\n\t</div>\n\t<div class=\"subheader primary-text-color\">\n\t\tLooks like we're having <br>\n\t\tsome server issues.\n\t</div>\n    <hr>\n    <div class=\"clearfix\"></div>\n    <div class=\"context primary-text-color\">\n\n    <p>\n      Go back to the previous page and try again.\n      If you think something is broken, report a problem.\n    </p>\n\n\n    </div>\n\t<div class=\"buttons-container\">\n\t\t<a class=\"border-button\" href=\"mailto:help@amplifr.com\" target=\"_blank\">Report A Problem</a>\n\t</div>\n</div>\n\n    <script>\n      function ErrorPage(e,t,n){this.$container=$(e),this.$contentContainer=this.$container.find(n==\"sign\"?\".sign-container\":\".content-container\"),this.pageType=t,this.templateName=n}ErrorPage.prototype.centerContent=function(){var e=this.$container.outerHeight(),t=this.$contentContainer.outerHeight(),n=(e-t)/2,r=this.templateName==\"sign\"?-100:0;this.$contentContainer.css(\"top\",n+r)},ErrorPage.prototype.initialize=function(){var e=this;this.centerContent(),this.$container.on(\"resize\",function(t){t.preventDefault(),t.stopPropagation(),e.centerContent()}),this.templateName==\"plain\"&&window.setTimeout(function(){e.$contentContainer.addClass(\"in\")},500),this.templateName==\"sign\"&&$(\".sign-container\").animate({textIndent:0},{step:function(e){$(this).css({transform:\"rotate(\"+e+\"deg)\",\"transform-origin\":\"top center\"})},duration:1e3,easing:\"easeOutBounce\"})},ErrorPage.prototype.createTimeRangeTag=function(e,t){return\"<time utime=\"+e+' simple_format=\"MMM DD, YYYY HH:mm\">'+e+\"</time> - <time utime=\"+t+' simple_format=\"MMM DD, YYYY HH:mm\">'+t+\"</time>.\"},ErrorPage.prototype.handleStatusFetchSuccess=function(e,t){if(e==\"503\")$(\"#replace-with-fetched-data\").html(t.status.description);else if(!t.scheduled_maintenances.length)$(\"#replace-with-fetched-data\").html(\"<em>(there are no active scheduled maintenances)</em>\");else{var n=t.scheduled_maintenances[0];$(\"#replace-with-fetched-data\").html(this.createTimeRangeTag(n.scheduled_for,n.scheduled_until)),$.fn.localizeTime()}},ErrorPage.prototype.handleStatusFetchFail=function(e){$(\"#replace-with-fetched-data\").html(\"<em>(enter a valid StatusPage.io url)</em>\")},ErrorPage.prototype.fetchStatus=function(e,t){if(!e||!t||t==\"404\")return;var n=\"\",r=this;t==\"503\"?n=e+\"/api/v2/status.json\":n=e+\"/api/v2/scheduled-maintenances/active.json\",$.ajax({type:\"GET\",url:n}).success(function(e,n){r.handleStatusFetchSuccess(t,e)}).fail(function(e,n){r.handleStatusFetchFail(t)})};\n      var ep = new ErrorPage('body', \"503\", \"plain\");\n      ep.initialize();\n\n      // hack to make sure content stays centered >_<\n      $(window).on('resize', function() {\n        $('body').trigger('resize')\n      });\n\n    </script>\n\n\n  </body>\n</html>

`
